### DISCUSSION TITLE
Possible to have InteractiveViewer start at the minimum scale?

### DISCUSSION THREAD

**Discussion by knightlygains** [flet-dev/flet#4542] [Q&A]
### Question

I'm using an InteractiveViewer in my app but it starts zoomed in a little more than I would like. Is there a way to have it start at the minimum zoom?

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
Can you share an example of what you mean? (Code, screen recording)

Also, we added some methods to the `InteractiveViewer` which you might find useful: https://flet.dev/docs/controls/interactiveviewer/#methods

**Reply by knightlygains** [reply]
https://github.com/user-attachments/assets/fbc04272-f5f6-4d6f-91b2-3dc11d255e54

Thanks for your quick response! In the video I show the InteractiveViewer code. I'm using it to host an image of a map of a building which I then place clickable icons representing "tickets" or "work orders" for my team. 
Anyways, you can see in the video that the interactive viewer starts with the content in the bottom right and it isn't at the minimum zoom. So to get the map to center, I have to zoom all the way out manually. I would like it to start the way it looks after I zoom all the way out.

I've tried the zoom method after the `page.add()` code and before but does not affect the zoom. The zoom method only seems to work after `ft.app()` has been run, so I am not sure how to programmatically run `zoom()`  without user intervention.

**Reply by knightlygains** [reply]
I updated the margins I was using, I moved it from the content to the boundary_margin parameter in the InteractiveViewer.
```
map_view = ft.InteractiveViewer(
        content=ft.Container(
        content=map_gesture_detector
    ),
    max_scale=15,
    min_scale=0.01,
    scale_factor=350,
    trackpad_scroll_causes_scale=True,
    clip_behavior=ft.ClipBehavior.ANTI_ALIAS,
    constrained=False,
    on_interaction_start=on_interaction_start,
    expand=True,
    boundary_margin=ft.margin.only(top=350, left=700, right=700, bottom=350)
)
```

This made the problem less severe, but the issue of the InteractiveViewer not starting at the minimum zoom still remains.

**Reply by ndonkoHenri** [reply]
`min_scale` is the minimum allowed scale of the viewer.

The `InteractiveViewer` defaults to rendering your content as-is, without taking into consideration the min or max scale values. Haven't found a way to modify that behaviour.
